services/device-registry-jdbc/src/main/java/org/eclipse/hono/deviceregistry/jdbc/impl/CredentialsServiceImpl.java

```diff
@@ -86,7 +90,7 @@ public CredentialsServiceImpl(final TableAdapterStore store) {
                             .put(CredentialsConstants.FIELD_AUTH_ID, key.getAuthId())
                             .put(CredentialsConstants.FIELD_SECRETS, new JsonArray(secrets));
 
-                    return CredentialsResult.from(HttpURLConnection.HTTP_OK, payload, CacheDirective.noCacheDirective());
+                    return CredentialsResult.from(HttpURLConnection.HTTP_OK, payload, ttl);
```

Comment:
The cache directive included in the response should be determined dynamically (based on the credentials type) as in the other registry implementations

Reply:
fixed